### PR TITLE
NEXT-8145 - minify compiled CSS fixes #65

### LINKED ISSUE #65: NEXT-8145 - CSS gets unminified by Autoprefixer
*EN:*

Description: 
 When you run theme:compile command in prod environment the css needs to be minified. 
 Within Storefront/Theme/ThemeCompiler.php the following line returns a minified css output

```
 $cssOutput = $this->scssCompiler->compile($variables . $concatenatedStyles);```

After that it puts the output into the Autoprefixer

```
 $autoPreFixer = new Autoprefixer($cssOutput);
 return $autoPreFixer->compile();
 ```

The Autoprefixer returns the css again however now with new lines included, so it isn't minified anymore.
 This is because of the used OutputFormat

```
 return $css_document->render(\Sabberworm\CSS\OutputFormat::createPretty());
 ```

With the OutputFormat to createCompact it would return the minfied version.

Steps to reproduce: 
 Run theme:compile and open the all.css of the theme.

Expected result: 
 A minified css.

Actual result: 
 A non-minified css :(

*DE:*

 

Beschreibung: 
 Wenn Sie den Befehl theme:compile in der Umgebung prod ausführen, muss das css minimiert werden. 
 In Storefront/Theme/ThemeCompiler.php gibt die folgende Zeile eine verkleinerte css-Ausgabe zurück

```
$cssOutput = $this->scssCompiler->compile($variables . $concatenatedStyles);

```

Danach legt es die Ausgabe in den Autoprefixer

```
$autoPreFixer = new Autoprefixer($cssOutput);
 return $autoPreFixer->compile();
 ```

Der Autoprefixer gibt das css wieder zurück, allerdings jetzt mit neuen Zeilen, so dass es nicht mehr verkleinert wird.
 Dies liegt an dem verwendeten OutputFormat

```
 return $css_document->render(\Sabberwurm-\CSS\OutputFormat::createPretty());
 ```

Mit dem OutputFormat zu createCompact würde es die verkleinerte Version zurückgeben.

Schritte zur Reproduktion: 
 theme:compile ausführen und die all.css des Themes öffnen.

Erwartetes Ergebnis: 
 Eine verkleinerte css.

Tatsächliches Ergebnis: 
 Ein nicht-minifiziertes css

### PR DESCRIPTION
**Pull request by aragon999**
### 1. Why is this change necessary?
Currently the CSS gets "deminified" by the autoprefixer, but there is currently no way to circumvent this, see:
https://github.com/padaliyajay/php-autoprefixer/pull/8
https://github.com/padaliyajay/php-autoprefixer/issues/10

### 2. What does this change do, exactly?
"Manually" minify the CSS in the CSS compiler.

### 3. Describe each step to reproduce the issue or behaviour.
Look at the compiled CSS with and without this PR.

### 4. Please link to the relevant issues (if any).
Fixes #65 

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Storefront/Theme/ThemeCompiler.php [hunk 1]
```diff
@@ -214,8 +216,20 @@ private function compileStyles(
             );
         }
         $autoPreFixer = new Autoprefixer($cssOutput);
-
-        return $autoPreFixer->compile();
+        $cssOutput = $autoPreFixer->compile();
```
Comment (shyim) [inline src/Storefront/Theme/ThemeCompiler.php:219]: I would prefer an own class which extends from https://github.com/padaliyajay/php-autoprefixer/blob/master/src/Autoprefixer.php#L16` and add this fix https://github.com/padaliyajay/php-autoprefixer/commit/475d41b7ee4213de6ce869fb8b58f384ebdd3186#diff-58df978638a78c3d9ed64d51ba83a775R22

Comment (tinect) [reply src/Storefront/Theme/ThemeCompiler.php:219]: I agree to @shyim cause this solution is slowing down compiling theme 🤔

Comment (aragon999) [reply src/Storefront/Theme/ThemeCompiler.php:219]: Mh, but when creating a child class of the autoprefixer, one either needs to do it twice as well, or implement everything what is in the base class?

I think this is more aimed that one does not forget it later on, right? And has an easier way to remove it later on.

File: changelog/_unreleased/2020-09-17-minify-css.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Minify CSS
+issue: NEXT-8145
+author: max
+author_email: max@swk-web.com
+author_github: aragon999
+---
+# Storefront
+*  Minify the compiled CSS
```
Comment (mitelg) [inline changelog/_unreleased/2020-09-17-minify-css.md:9]: this is not a valid changelog entry :blush:

File: src/Storefront/Theme/ThemeCompiler.php [hunk 3]
```diff
@@ -214,8 +213,7 @@ private function compileStyles(
             );
         }
         $autoPreFixer = new Autoprefixer($cssOutput);
-
-        return $autoPreFixer->compile();
+        return $autoPreFixer->compile(false);
```
Comment (tinect) [inline src/Storefront/Theme/ThemeCompiler.php:216]: This causes minification also on debug mode, doesn't it? You could respect the `$debug` or scssCompiler::Formatter 🤔

**Review by shyim — APPROVED** [review]

**Pull request merged**